File: solutions/python/01/gcd_fn/main.py
import sys


def gcd(a: int, b: int) -> int:
    while b:
        a, b = b, a % b
    return a


def main() -> None:
    a, b = map(int, sys.stdin.readline().split())
    print(gcd(a, b))


if __name__ == "__main__":
    main()

File: solutions/go/01/gcd_fn/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func gcd(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

func main() {
	in := bufio.NewReader(os.Stdin)
	var a, b int
	fmt.Fscan(in, &a, &b)
	fmt.Println(gcd(a, b))
}
